src/openrct2/world/TileInspector.cpp

```diff
@@ -742,6 +742,32 @@ GameActionResult::Ptr tile_inspector_wall_set_slope(
     return std::make_unique<GameActionResult>();
 }
 
+GameActionResult::Ptr tile_inspector_wall_animation_frame_offset(
+    const CoordsXY& loc, int16_t elementIndex, int8_t animationFrameOffset, bool isExecuting)
+{
+    TileElement* const wallElement = map_get_nth_element_at(loc, elementIndex);
+
+    if (wallElement == nullptr || wallElement->GetType() != TILE_ELEMENT_TYPE_WALL)
+        return std::make_unique<GameActionResult>(GA_ERROR::UNKNOWN, STR_NONE);
+
+    if (isExecuting)
+    {
+        uint8_t animationFrame = wallElement->AsWall()->GetAnimationFrame();
+        wallElement->AsWall()->SetAnimationFrame(animationFrame + animationFrameOffset);
+
+        map_invalidate_tile_full(loc);
+
+        rct_window* const tileInspectorWindow = window_find_by_class(WC_TILE_INSPECTOR);
+        if (tileInspectorWindow != nullptr && (uint32_t)(loc.x / 32) == windowTileInspectorTileX
```

Comment:
I assume you have copied this from somewhere else? I don't see the point in performing all these checks I would just invalidate the TileInspector not much to be gained from checking that it is hovering over the correct tile.

Reply:
Yes, the other tile inspector game actions do it as well, it's meant for multiplayer, so it only invalidates the tile inspector windows for the users that have the tile selected.